src/component/feature/progress.jsx

```diff
@@ -23,16 +24,78 @@ class Progress extends Component {
     }
 
     componentWillReceiveProps ({ percentage }) {
-        this.setState({ percentage });
+        if (this.state.percentage !== percentage) {
+            const nextState = { percentage };
+            if (this.props.animatePercentageText) {
+                this.animateTo(percentage, this.getTarget(percentage));
+            } else {
+                nextState.percentageText = percentage;
+            }
+            this.setState(nextState);
+        }
+    }
+
+    getTarget (target) {
```

Comment:
Cannot reproduce that big of a difference here, I know there is some - as the wheel is css-animation...